diff --git a/gdbserver/configure.srv b/gdbserver/configure.srv
index 0cb5072c8ab..a88404e777a 100644
--- a/gdbserver/configure.srv
+++ b/gdbserver/configure.srv
@@ -72,6 +72,7 @@ case "${gdbserver_host}" in
 			srv_linux_usrregs=yes
 			srv_linux_thread_db=yes
 			;;
+  arm*-*-uclinux* | \
   arm*-*-linux*)	srv_tgtobj="$srv_linux_obj linux-arm-low.o"
 			srv_tgtobj="$srv_tgtobj linux-arm-tdesc.o"
 			srv_tgtobj="$srv_tgtobj linux-aarch32-low.o"
